htmd/builder/charmm.py

```diff
@@ -251,12 +250,29 @@ def build(mol, topo=None, param=None, stream=None, prefix='structure', outdir='.
         f.write('coordpdb ' + path.join('segments', pdbname) + ' ' + seg + '\n\n')
 
     # Printing out patches for the disulfide bridges
+    # TODO: Remove this once we deprecate the class
+    from htmd.builder.builder import DisulfideBridge
+    from htmd.molecule.molecule import UniqueResidueID
+    if disulfide is not None and len(disulfide) != 0 and isinstance(disulfide[0], DisulfideBridge):
+        newdisu = []
+        for d in disulfide:
+            r1 = UniqueResidueID.fromMolecule(mol, 'resid {} and segname {}'.format(d.resid1, d.segid1))
+            r2 = UniqueResidueID.fromMolecule(mol, 'resid {} and segname {}'.format(d.resid2, d.segid2))
+            newdisu.append([r1, r2])
+        disulfide = newdisu
+    # TODO: Remove up to here ----------------------
+
+    if disulfide is not None and len(disulfide) != 0 and isinstance(disulfide[0][0], str):
+        disulfide = convertDisulfide(mol, disulfide)
+
     if disulfide is None:
         disulfide = detectDisulfideBonds(mol)
 
     if len(disulfide) != 0:
         for d in disulfide:
-            f.write('patch DISU {}:{} {}:{}\n'.format(d.segid1, d.resid1, d.segid2, d.resid2))
+            str0 = '{}:{}{}'.format(d[0].segid, d[0].resid, d[0].insertion)
```

Comment:
This works in CHARMM? Cool!

Reply:
Yes! I even tested it and it worked. I was also surprised :P